### PR TITLE
changed the way we get the colors on vehicle spawn

### PR DESCRIPTION
**Pull request by Olrosse**
using getField had issues where the base color could return as nil, this should hopefully fix that by using getMetallicPaintData and veh.color, veh.colorPalette etc instead.

I also removed a duplicate of getObjectByID

### CODE REVIEW

File: lua/ge/extensions/MPVehicleGE.lua [hunk 1]
```diff
@@ -924,7 +923,10 @@ local function sendVehicleSpawn(gameVehicleID)
 		vehicleTable.rot = {rot.x, rot.y, rot.z, rot.w} -- Rotation
 		
 		-- The vehicle_manager.lua may not contain the correct color values, since v0.31, when we read them from that lua, so we read those from the object itself
-		vehicleTable.vcf.paints = MPHelpers.getColorsFromVehObj(veh)
+		local metallicPaintData = veh:getMetallicPaintData()
+		vehicleTable.vcf.paints[1] = createVehiclePaint(veh.color, metallicPaintData[1])
+		vehicleTable.vcf.paints[2] = createVehiclePaint(veh.colorPalette0, metallicPaintData[2])
+		vehicleTable.vcf.paints[3] = createVehiclePaint(veh.colorPalette1, metallicPaintData[3])
```
Comment (OfficialLambdax) [inline lua/ge/extensions/MPVehicleGE.lua]: I think this should be refactored into the `MPHelpers.getColorsFromVehObj()` function. Point is that atleast one more pr is building on top of it #560

**Review by OfficialLambdax — CHANGES_REQUESTED** [review]

File: lua/ge/extensions/MPVehicleGE.lua (hunk 1 above)
Comment (Olrosse) [reply lua/ge/extensions/MPVehicleGE.lua]: I'll change it once I'm on, i was originally thinking we could just do the same for that PR as well but yeah having it as one function makes more sense

**Comment by OfficialLambdax** [comment]
Tests
- [x] Spawn a vanilla config with its default color
- [x] Spawn a vanilla config with a custom preconfigured color
- [x] Spawn a vanilla config with a entirely custom color
- [x] Edit into a vanilla config with its default color
- [x] Edit into a vanilla config with a custom preconfigured color
- [x] Edit into a vanilla config with a entirely custom color
- [x] Spawn a snowman unicycle
- [x] Spawn a custom config with its default color
- [ ] Spawn a custom config with a custom preconfigured color
- [ ] Spawn a custom config with a entirely custom color
- [x] Edit into a custom config with its default color
- [ ] Edit into a custom config with a custom preconfigured color
- [ ] Edit into a custom config with a entirely custom color
- [x] 1) Spawn vanilla config with default color 2) Edit to the same vehicle but with a custom config
- [x] Spawn the config from this issue https://github.com/BeamMP/BeamMP/issues/572

**Comment by SaltySnail** [comment]
The unmarked tests don't work in vanilla beamng for certain configs, but the colors are the same between clients.

**Review by OfficialLambdax — APPROVED** [review]
Approving and merging on the basis of the test results from the two dev team members. Can confirm that the missing checkmarks are based on game issues.

Thanks @Olrosse for the pr. Thanks @SaltySnail for helping in testin (:

**Pull request merged**